### PR TITLE
Replaced files, shell with app_instance in PackageDatabase methods and tests

### PR DESCRIPTION
**Pull request by schmerl**
This is number 1 of the following changes:

1. Change package tests to use sut (app_instance)
2. Change PackageDatabase to use build
3. Separate PD into PD <- PDR1 & <- PDR2

### CODE REVIEW

File: src/roswire/app/description.py [hunk 1]
```diff
@@ -109,11 +109,9 @@ def path(cls, app: 'App') -> str:
     def for_app(cls, app: 'App') -> 'AppDescription':
         """Produces a description for a given application."""
         with app.launch(require_description=False) as app_instance:
-            shell = app_instance.shell
-            files = app_instance.files
-            paths = PackageDatabase.paths(shell, files)
-            db_package = PackageDatabase.from_paths(files, paths)
-            distribution_name = shell.environ('ROS_DISTRO')
+            paths = PackageDatabase.paths(app_instance)
```
Comment (schmerl) [inline src/roswire/app/description.py:112]: This will merge into build in the next change.

File: src/roswire/common/package.py [hunk 2]
```diff
@@ -104,12 +109,13 @@ class PackageDatabase(Mapping[str, Package]):
         :class:`dict` operations are provided (e.g., :code:`del db['foo'])`
         and `db['foo'] = bar`).
     """
+
```
Comment (ChrisTimperley) [inline src/roswire/common/package.py:112]: extra space

File: src/roswire/common/package.py [hunk 3]
```diff
@@ -139,18 +144,19 @@ def _paths_ros2(shell: dockerblade.Shell,
...
-        distro = shell.environ('ROS_DISTRO')
-        if distro in {'dashing', 'eloquent', 'foxy'}:
-            return cls._paths_ros2(shell, files)
-        return cls._paths_ros1(shell, files)
+    def paths(cls, app_instance: 'AppInstance') -> List[str]:
+        """Parses :code:`ROS_PACKAGE_PATH` for a given shell.
+
+        Parameters
+        ----------
+        app_instance; The App instance for the image
```
Comment (ChrisTimperley) [inline src/roswire/common/package.py:152]: Doc string format is:
```
app_instance: AppInstance
  An instance of the application for which a list of paths should be obtained.
```

File: src/roswire/common/package.py [hunk 4]
```diff
@@ -160,7 +166,7 @@ def from_paths(files: dockerblade.FileSystem,
 
         Parameters
         ----------
-        files: dockerblade.FileSystem
+        app_instance: AppInstance
             access to the filesystem.
```
Comment (ChrisTimperley) [inline src/roswire/common/package.py:170]: Doc string above is correct, but this line is no longer accurate

File: test/test_package.py [hunk 5]
```diff
@@ -48,7 +48,7 @@ def test_build(filesystem):
                             'name': 'dot_graph'}]}}
         ]
     })
-    actual = Package.build(path, filesystem)
+    actual = Package.build(path, sut.files)
```
Comment (ChrisTimperley) [inline test/test_package.py:51]: This should be just `sut`

File: test/test_package.py [hunk 6]
```diff
@@ -237,7 +237,7 @@ def test_database_paths(sut):
         '/opt/ros/melodic/share/xacro',
         '/opt/ros/melodic/share/xmlrpcpp'
     }
-    actual = set(PackageDatabase.paths(sut.shell, sut.files))
+    actual = set(PackageDatabase.paths(undefined_variable))
```
Comment (ChrisTimperley) [inline test/test_package.py:240]: Should be `sut`

File: test/test_package.py [hunk 7]
```diff
@@ -431,7 +431,7 @@ def test_package_location_ros2(sut):
                       '/ros_ws/install/launch',
                       '/ros_ws/install/ament_cmake_export_link_flags'
                       }
-    actual_paths = set(PackageDatabase.paths(sut.shell, sut.files))
+    actual_paths = set(PackageDatabase.paths(undefined_variable))
```
Comment (ChrisTimperley) [inline test/test_package.py:434]: Should be `sut`

File: src/roswire/common/package.py (hunk 2 above)
Comment (schmerl) [reply src/roswire/common/package.py]: Fixed

File: src/roswire/common/package.py (hunk 3 above)
Comment (schmerl) [reply src/roswire/common/package.py]: Fixed

File: src/roswire/common/package.py (hunk 4 above)
Comment (schmerl) [reply src/roswire/common/package.py]: Fixed

File: test/test_package.py (hunk 5 above)
Comment (schmerl) [reply test/test_package.py]: Done, and in other tests too

File: test/test_package.py (hunk 6 above)
Comment (ChrisTimperley) [reply test/test_package.py:240]: @schmerl This seems to be causing the test failure

File: test/test_package.py (hunk 7 above)
Comment (ChrisTimperley) [reply test/test_package.py]: @schmerl This seems to be causing the test failure

**Review by ChrisTimperley — APPROVED** [review]

**Pull request merged**